### DISCUSSION TITLE
Cannot run app on Win7 - ssl error

### DISCUSSION THREAD

**Discussion by pmonsky** [cefsharp/CefSharp#4008] [General]
Hi,
we are currently facing a strange problem on one computer with our win application. When trying to interact with UI we get the following error:
_[0215/150113.099:ERROR:ssl_client_socket_impl.cc(960)] handshake failed; returned -1, SSL error code 1, net_error -101_
The application is installed on many computers (Win7, Win8.1, Win10, Win11), but there are few cases with this error, so the application is completely unusable.
I have only this error message in debug.log file. I have checked the certificates (self-signed - SSL communication over localhost) and it seems to be ok.
What else do I have to check? I have no idea what is wrong with these specific cases. In most cases, on Win7 the application works fine.
Chromium version is: 86.

**Comment by amaitland** [comment]
As it's a error coming from the chromium network stack I'd suggest trying a newer version. Version 86 is old and unsupported. 

For testing purposes you can use https://github.com/cefsharp/CefSharp.MinimalExample

**Reply by pmonsky** [reply]
Ok, updating chromium version to 98 did not help. The same error still occurs:

_[0217/093656.049:ERROR:ssl_client_socket_impl.cc(996)] handshake failed; returned -1, SSL error code 1, net_error -101_

**Reply by amaitland** [reply]
You'll need to compare a working Windows 7 machine to your non working one.

Check firewall/antivirus, make sure fully patched. 
Check the website loads in chrome 98
Check the chromium issue tracker for the same error. 

The error by itself doesn't yield anything particular meaningful. 

https://source.chromium.org/chromium/chromium/src/+/refs/tags/98.0.4758.80:net/socket/ssl_client_socket_impl.cc;l=996

**Reply by pmonsky** [reply]
I cannot post the addres here because it is a local address _https://localhost:[port_number]/method_
I have checked it on 4 browsers (IE, FF, Chrome, Edge) on non working machine:
1. IE 11 - OK
2. FF (97.0) - OK
3. Chrome (98.0.4758.102) - NOT OK (ERR_CONNECTION_RESET) 
4. MS Edge - NOT OK (ERR_CONNECTION_RESET)

**Reply by amaitland** [reply]
Does a different working win7 machine load the website in Chrome/Edge(Chromium)?

Have you triple checked the certificate is correct?  Delete and create a new one? 

What's different about the non working machine? Antivirus? Patches? Configured by someone else?

**Reply by pmonsky** [reply]
> Does a different working win7 machine load the website in Chrome/Edge(Chromium)?

yes, on a working case on other Win7, website content loads on Chromium/Edge without any error.


> Have you triple checked the certificate is correct? Delete and create a new one?

yes, moreover, on non working machine, when loading website in IE/FF I see green padlock in address bar.


> What's different about the non working machine? Antivirus? Patches? Configured by someone else?

hard to say. It is surely configured by someone else. There are AV and firewall, but when it all is turned off nothing changes.
I don't even know which components of the system/registry we have to compare on working and non working examples.
